src/cunumeric/sort/searchsorted.cc

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/sort/searchsorted.h"
+#include "cunumeric/sort/searchsorted_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <LegateTypeCode CODE>
+struct SearchSortedImplBody<VariantKind::CPU, CODE> {
+  using VAL = legate_type_of<CODE>;
+
+  void operator()(const Array& input_array,
+                  const Array& input_values,
+                  const Array& output_positions,
+                  const Rect<1>& rect_base,
+                  const Rect<1>& rect_values,
+                  const bool left,
+                  const bool is_index_space,
+                  const size_t volume,
+                  const int64_t global_volume,
+                  const size_t num_values)
+  {
+    auto input   = input_array.read_accessor<VAL, 1>(rect_base);
+    auto input_v = input_values.read_accessor<VAL, 1>(rect_values);
+    assert(input.accessor.is_dense_arbitrary(rect_base));
```

Comment:
please request an exact instance for the input to make this assertion hold all the time